contrib/src/databases.rs

```diff
@@ -0,0 +1,84 @@
+pub extern crate r2d2;
+
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::io;
+use std::marker::{Send, Sized};
+use rocket::config::{self, ConfigError, Value};
+
+use self::r2d2::ManageConnection;
+
+#[cfg(feature = "diesel_sqlite")]
+pub extern crate diesel;
+
+#[cfg(feature = "pg_pool")]
+pub extern crate postgres;
+#[cfg(feature = "pg_pool")]
+pub extern crate r2d2_postgres;
+
+pub struct DatabaseConfig<'a> {
+    pub url: &'a str,
+    pub pool_size: u32,
+    pub extras: Option<&'a BTreeMap<String, Value>>,
+}
+
+pub fn database_config<'a>(name: &str, from: &'a config::Config) -> Option<DatabaseConfig<'a>> {
+    from.get_slice("databases").and_then(|db_config| {
+        db_config
+            .iter()
+            .filter(|element|
+                match element.as_table().and_then(|table|
+                    Some(table["name"].as_str() == Some(name))
+                ) {
+                    Some(true) => true,
+                    _ => false,
+                }
+            )
+            .nth(0)
+            .ok_or(ConfigError::UnknownKey(format!("Could not find key: {}", name)))
+    })
+    .and_then(|connection_config| {
+        Ok(connection_config["url"].as_str().and_then(|url| {
+            Some(DatabaseConfig {
+                url,
+                pool_size: connection_config["pool_size"].as_integer().unwrap_or(from.workers as i64) as u32,
+                extras: connection_config.as_table(),
+            })
+        }))
+    })
+    .ok()
+    .and_then(|result| result)
+}
+
+pub trait Poolable: Send + Sized + 'static {
+    type Manager: ManageConnection<Connection=Self>;
+    type Error;
+
+    fn manager(config: &DatabaseConfig) -> Result<Self::Manager, Self::Error>;
+}
+
+#[cfg(feature = "diesel_sqlite")]
+impl Poolable for diesel::SqliteConnection {
+    type Manager = diesel::r2d2::ConnectionManager<diesel::SqliteConnection>;
+    type Error = io::Error;
+
+    fn manager(info: &DatabaseConfig) -> Result<Self::Manager, Self::Error> {
+        Ok(diesel::r2d2::ConnectionManager::new(info.url.clone()))
```

Comment:
You don't need to `clone()` here since `new()` takes anything that's `Into<String>`. Better to leave it to the function to do what it wants.

Reply:
Good to know. I'll drop the `clone()` call.